ou_dedetai/config.py

```diff
@@ -1,209 +1,1052 @@
-        except FileNotFoundError:
-            logging.info(f"No config file not found at {config_file_path}")
-            return config_dict
-        except json.JSONDecodeError as e:
-            logging.error("Config file could not be read.")
-            logging.error(e)
+
+    @property
+    def install_dir_default(self) -> str:
+        return f"{str(Path.home())}/{self.faithlife_product}Bible{self.faithlife_product_version}"  # noqa: E501
+
+    @property
+    def install_dir(self) -> str:
+        if self._overrides.install_dir:
+            return self._overrides.install_dir
+        default = self.install_dir_default
+        question = f"Where should {self.faithlife_product} files be installed to?: "  # noqa: E501
+        options = [default, PROMPT_OPTION_DIRECTORY]
+        output = self._ask_if_not_found("install_dir", question, options)
+        return output
+    
+    @install_dir.setter
+    def install_dir(self, value: str | Path):
+        value = str(Path(value).absolute())
+        if self._raw.install_dir != value:
+            self._raw.install_dir = value
+            # Reset cache that depends on install_dir
+            self._wine_appimage_files = None
+            self._write()
+
+    @property
+    # This used to be called APPDIR_BINDIR
+    def installer_binary_dir(self) -> str:
+        if self._overrides.installer_binary_dir is not None:
+            return self._overrides.installer_binary_dir
+        return f"{self.install_dir}/data/bin"
+
+    @property
+    # This used to be called WINEPREFIX
+    def wine_prefix(self) -> str:
+        if self._overrides.wine_prefix is not None:
+            return self._overrides.wine_prefix
+        return get_wine_prefix_path(self.install_dir)
+
+    @property
+    def wine_binary(self) -> str:
+        """Returns absolute path to the wine binary"""
+        output = self._raw.wine_binary
+        if output is None:
+            question = f"Which Wine AppImage or binary should the script use to install {self.faithlife_product} v{self.faithlife_product_version} in {self.install_dir}?: "  # noqa: E501
+            options = utils.get_wine_options(self.app)
+
+            choice = self.app.ask(question, options)
+
+            output = choice
+            self.wine_binary = choice
+        # Return the full path so we the callee doesn't need to think about it
+        if self._raw.wine_binary is not None and not Path(self._raw.wine_binary).exists() and (Path(self.install_dir) / self._raw.wine_binary).exists(): # noqa: E501
+            return str(Path(self.install_dir) / self._raw.wine_binary)
+        return output
+
+    @wine_binary.setter
+    def wine_binary(self, value: str):
+        """Takes in a path to the wine binary and stores it as relative for storage"""
+        # Make the path absolute for comparison
+        relative = self._relative_from_install_dir(value)
+        # FIXME: consider this, it doesn't work at present as the wine_binary may be an
```

Comment:
Relative paths and appimage are used so as not to break the appimage symlinks in the event the installer folder is moved after install